hermes-common/src/main/java/pl/allegro/tech/hermes/common/metric/ProducerMetrics.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    public double getBufferTotalBytes() {
+        return meterRegistry.get(ACK_ALL_BUFFER_TOTAL_BYTES).gauge().value()
+                + meterRegistry.get(ACK_LEADER_BUFFER_TOTAL_BYTES).gauge().value();
+    }
+
+    public double getBufferAvailableBytes() {
+        return meterRegistry.get(ACK_ALL_BUFFER_AVAILABLE_BYTES).gauge().value()
+                + meterRegistry.get(ACK_LEADER_BUFFER_AVAILABLE_BYTES).gauge().value();
+    }
+
+    public <T> void registerProducerInflightRequestGauge(T stateObj, ToDoubleFunction<T> f) {
+        meterRegistry.gauge(INFLIGHT_REQUESTS, stateObj, f);
+        hermesMetrics.registerProducerInflightRequest(() -> (int) f.applyAsDouble(stateObj));
+    }
+
+    public <T> void registerAckAllMaxLatencyBrokerGauge(T stateObj, ToDoubleFunction<T> f, String brokerNodeId) {
+        registerLatencyPerBrokerGauge(stateObj, f, "request-latency-max", ACK_ALL, brokerNodeId);
+    }
+
+    public <T> void registerAckLeaderMaxLatencyPerBrokerGauge(T stateObj, ToDoubleFunction<T> f, String brokerNodeId) {
+        registerLatencyPerBrokerGauge(stateObj, f, "request-latency-max", ACK_LEADER, brokerNodeId);
+    }
+
+    public <T> void registerAckAllAvgLatencyPerBrokerGauge(T stateObj, ToDoubleFunction<T> f, String brokerNodeId) {
+        registerLatencyPerBrokerGauge(stateObj, f, "request-latency-avg", ACK_ALL, brokerNodeId);
+    }
+
+    public <T> void registerAckLeaderAvgLatencyPerBrokerGauge(T stateObj, ToDoubleFunction<T> f, String brokerNodeId) {
+        registerLatencyPerBrokerGauge(stateObj, f, "request-latency-avg", ACK_LEADER, brokerNodeId);
+    }
+
+    private <T> void registerLatencyPerBrokerGauge(T stateObj,
+                                                   ToDoubleFunction<T> f,
+                                                   String metricName,
+                                                   String producerName,
+                                                   String brokerNodeId) {
+        String baseMetricName = KAFKA_PRODUCER + producerName +  metricName;
+        String graphiteMetricName = baseMetricName + "." + escapeDots(brokerNodeId);
+
+        gaugeRegistrar.registerGauge(
+                graphiteMetricName, baseMetricName, stateObj, f, Tags.of("broker", brokerNodeId)
+        );
+    }
+
+    private static final String KAFKA_PRODUCER = "kafka-producer.";
+    private static final String ACK_LEADER = "ack-leader.";
+    private static final String ACK_ALL = "ack-all.";
+
+    private static final String ACK_ALL_BUFFER_TOTAL_BYTES = KAFKA_PRODUCER + ACK_ALL + "buffer-total-bytes";
+    private static final String ACK_ALL_BUFFER_AVAILABLE_BYTES = KAFKA_PRODUCER + ACK_ALL + "buffer-available-bytes";
+    private static final String ACK_ALL_METADATA_AGE = KAFKA_PRODUCER + ACK_ALL + "metadata-age";
+    private static final String ACK_ALL_RECORD_QUEUE_TIME_MAX = KAFKA_PRODUCER + ACK_ALL + "record-queue-time-max";
```

Comment:
They are, see: https://github.com/allegro/hermes/pull/1695/files/dd3c91fb565e3a1fe2bc44408d6b57a668d6e3f7#diff-e9fd10aab35b3d7b1bedf1d9f0249c2fe5a1f0cc874c4849041f5c522159b180R62

Reply:
Oh, right, I overlooked it. However, I noticed that sometimes you use `milliseconds` as a unit, which is not in line with what the [documentation on best practices](https://prometheus.io/docs/practices/naming/) says:

> ...should use base units (e.g. seconds, bytes, meters - not milliseconds, megabytes, kilometers).

Yesterday, I discovered this API: 
```java
meterRegistry.more().timeGauge("consumers-rate.max-rate.coordinator.duration", List.of(), obj, TimeUnit.MILLISECONDS, f);
```

I used it in consumers, but I haven't tested it yet. I'm going to do so today.

Reply:
I thought this best practice does not apply to Gauges (because we would need to scale them accordingly), but the API you mentioned is exactly what we needed! [Docs](https://micrometer.io/docs/concepts#_timegauge) say that the metric will automatically be converted to seconds and that the name suffix will be added, very cool